modules/loader/src/main/java/com/opengamma/strata/loader/CharSources.java

```diff
@@ -0,0 +1,130 @@
+  }
+
+  /**
+   * Obtains an instance of {@link CharSource} from a file name, specified as a String. This also takes in a specific character set, as a {@link Charset}.
+   *
+   * @param fileName the file name, as a String
+   * @param charset the charset to build the new CharSource based on
+   * @return a new instance of {@link CharSource}
+   */
+  public static CharSource ofFileName(String fileName, Charset charset) {
+    return Files.asCharSource(new File(fileName), charset);
+  }
+
+  //---------------------------------------------------------------------------------------------
+  /**
+   * Obtains an instance of {@link CharSource} from a file object, specified as a {@link File}.
+   *
+   * @param file the file object
+   * @return a new instance of {@link CharSource} with the default charset on the current JVM.
+   */
+  public static CharSource ofFile(File file) {
+    return Files.asCharSource(file, Charsets.UTF_8);
+  }
+
+  /**
+   * Obtains an instance of {@link CharSource} from a file object, specified as a {@link File}. This also takes in a specific character set, as a {@link Charset}.
+   *
+   * @param file the file object
+   * @param charset the charset to build the new CharSource based on
+   * @return a new instance of {@link CharSource}
+   */
+  public static CharSource ofFile(File file, Charset charset) {
+    return Files.asCharSource(file, charset);
+  }
+
+  //---------------------------------------------------------------------------------------------
+  /**
+   * Obtains an instance of {@link CharSource} from a file path, specified as a {@link Path}.
+   *
+   * @param path the path to create a {@link CharSource} from
+   * @return a new instance of {@link CharSource} with the default charset on the current JVM.
+   */
+  public static CharSource ofPath(Path path) {
+    return Files.asCharSource(path.toFile(), Charsets.UTF_8);
+  }
+
+  /**
+   * Obtains an instance of {@link CharSource} from a file path, specified as a {@link Path}. This also takes in a specific character set, as a {@link Charset}.
+   *
+   * @param path the path to create a {@link CharSource} from
+   * @param charset the charset to build the new CharSource based on
+   * @return a new instance of {@link CharSource}
+   */
+  public static CharSource ofPath(Path path, Charset charset) {
+    return Files.asCharSource(path.toFile(), charset);
+  }
+
+  //---------------------------------------------------------------------------------------------
+  /**
+   * Obtains an instance of {@link CharSource} from a resource such as a file specified as a {@link ResourceLocator}. This also takes in a specific character set, as a {@link Charset}.
+   *
+   * @param resourceLocator the resource to create a {@link CharSource} for, as an instance of {@link ResourceLocator}
+   * @return a new instance of {@link CharSource} with the default charset on the current JVM.
+   */
+  public static CharSource ofResourceLocator(ResourceLocator resourceLocator) {
```

Comment:
Looking at `ResourceLocator` and `ByteSource`, it seems like adding the methods here encourages use of a static class when we should be encouraging users to call the conversion methods themselves.

What would be useful would be `of(URL)` `ofContent(String)` and  `ofContent(byte[], CharSet)`.